microsetta_admin/templates/manage_projects.html

```diff
@@ -0,0 +1,602 @@
+    const dispositionGroupColNums = fillRange(15, 18);
+    const sampleTypeGroupColNums = fillRange(19, 23);
+    const profilingGroupColNums = fillRange(24, 34);
+    const brandingGroupColNums = fillRange(35, 36);
+    const kitsGroupColNums = fillRange(37, 41);
+    const samplesGroupColNums = fillRange(42, 48);
+
+    function showAllCols(){
+        projectTable.columns().visible(true);
+    }
+    // end datatable set-up
+
+    // functions for resetting, populating, and displaying the
+    // create/update modal
+    function resetModal(){
+        showAllCols();
+        document.getElementById("project_form").reset();
+    }
+
+    function readRow(projId){
+        showAllCols();
+
+        let projIdSuffix = "_" + projId;
+        let projRowIdSelector = "#tr_project" + projIdSuffix;
+        $(projRowIdSelector).children('td').each(function () {
+            // any non-data columns (like edit button column)
+            // don't have an id, so we know to skip them
+            if (this.id) {
+                // get the form element in the modal edit form
+                // that has the same base id as the data cell we're examining
+                let formElmId = this.id.replace("td_", "");
+                formElmId = formElmId.replace(projIdSuffix, "");
+                let formElm = $("#" + formElmId);
+
+                // if we got an editable element with this id (which we
+                // won't always--e.g., no editable element for num_received_samples
+                // bc that is read-only from db)
+                if (formElm.length === 1){
+                    // if the element is a checkbox, it needs special
+                    // finicky handling to set its value
+                    if (formElm.is(":checkbox")) {
+                        // set checked property to boolean value of data field
+                        let isChecked = (this.innerText.toLowerCase() === "true");
+                        formElm.prop('checked', isChecked);
+                    } else {
+                        // set the value of the form element to the value
+                        // of the relevant cell in the projects table
+                        formElm.val(this.innerText);
+                    }
+                }
+            }
+        });
+    }
+
+    // called when the modal submit is clicked (NOT at onsubmit, which is too late)
+    function unhideAllPanels() {
+        // make all the panels visible.  This is a kludge needed because input elements
+        // that are not visible are not included when the form is posted :(
+        $(".collapse").addClass("show").css("height","auto");
+        // disable the submit button
+        $('#modal_submit').addClass('disabled');
+        // show a wait message since the page reload is not instant
+        $('#wait_msg').show();
+    }
+
+    $(document).ready(function() {
+        document.getElementById('bank_samples').onchange = function() {
```

Comment:
Are jquery onchange functions called immediately upon registration, or could this be out of sync when the page is first loaded?

Reply:
Damn your devilish attention to detail! :D Yes, indeed, it COULD have been out of sync ... I have added a manual call to the onchange handler (which, note, does not *assume* a change happened but rather acts on whatever the current status of the checkbox is) in `readRow` to ensure that if the box is checked, the date input is NOT disabled.